libwidgets/Chrome/BreadcrumbElement.vala

```diff
@@ -197,6 +198,10 @@ public class Marlin.View.Chrome.BreadcrumbElement : Object {
             icon_to_draw = PF.PixbufUtils.lucent (icon_to_draw, 50);
         }
 
+        if (offset_px >= width) {
+            room_for_text = false;
```

Comment:
Hm, and this is not correct, I've missed icon here.

Reply:
I've redone this with masking. See commit descriptions for details, there's sth weird going on with clipping + `render_layout()`.